### ISSUE TITLE
Cannot execute configuration 

### ISSUE THREAD

**Issue by NAVRockClimber** [music-assistant/hass-music-assistant#637]
### What version of Music Assistant has the issue?

2022.7.2

### The problem

I suspect I got the same issue as #635 but my error message differs. After the download I receive the following message if I try to add the integration.

![image](https://user-images.githubusercontent.com/37535503/178157934-6a809c12-7777-4f08-8a4c-98176ef0c72c.png)

### How to reproduce

Not sure I can provide all parameters but donwload music assistant via HACS, restart Home Assistant, add integration. After a few seconds the error message above appears

### Relevant log output

2022-07-10 20:41:01 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading configuration flow for integration mass: No module named 'music_assistant'

### Additional information

running home assistant as container in a kubernetes cluster

### What version of Home Assistant Core are your running

2022.7.2

### What type of installation are you running?

Home Assistant Core

### On what type of hardware are you running?

Alternative

**Comment by marcelveldt** [comment]
There is no version 2022.7.2 of MA so can you please tell me exactly what version you're running ?
Did you install with HACS ?
You're running core yourself... What architecture and OS ?
Note that running core stand-alone is considered a very advanced setup only recommended for development setups or when you really know what you're doing. For example you will have to make sure that the correct ports are open and a 
recent version of ffmpeg installed etc

**Comment by NAVRockClimber** [comment]
Sorry, my brain took the wrong road and I entered accidently the HA version. 

I tried the versions: 2022.6.10 and 2022.7 BETA 3
Yes, I used HACS

Running the official [container image](https://hub.docker.com/r/homeassistant/home-assistant) in a Kubernetes cluster. 
I am an IT professional for 18 years. I know which ports to open. ;)

**Comment by marcelveldt** [comment]
So you're running container and not the bare core. That's better ;-)
That should work perfectly fine out of the box and I've never seen this error before. Maybe just the browser acting up ? Can you try to clear the cache of the browser and/or restart HA ?

**Comment by NAVRockClimber** [comment]
I just tried all browsers I got installed brave (chromium based), chrome, edge.

I suspect it's the virtualization? Sounds similar to issue #635. Anything next to ffmpeg I can check?

**Comment by marcelveldt** [comment]
Can you please try to re-install ?
It's complaining about the music_assistant library missing somehow so something is messed up with the install somehow.
Does installing the stable version 2022.6.x work ?

**Comment by NAVRockClimber** [comment]
I tried the first after the beta failed as this was my first thought. 
I tried again and re-installed 2022.6.10 but the problem stays. Where is the library installed to? 

I removed my gitignore temporarely and git shows me the following files/folders:

![image](https://user-images.githubusercontent.com/37535503/178160445-3c5e89f4-6d6a-4e96-bf1e-6a2969516408.png)

**Comment by marcelveldt** [comment]
the library is installed as requirement by HA itself. You are sure that you only see an error about the module 'music_assistant' can not be found? Nothing else relevant in there ?

This should work out of the box, especially when you're running the HA OS or the official HA docker image.
Strange...

Maybe somehow the installation of the music_assistant library failed due to some conflict or something.
In that case I hope the log will have some answer.

**Comment by NAVRockClimber** [comment]
This is my whole log:

2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret my_latitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret my_longitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Loading /config/packages/persistence/secrets.yaml
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Loading /config/packages/secrets.yaml
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret mariadb_url retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Loading /config/packages/presence/secrets.yaml
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret work_latitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret work_longitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret my_latitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret my_longitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret pueppi_latitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret pueppi_longitude retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Loading /config/packages/weather/secrets.yaml
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret darksky_api_key retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Loading /config/packages/misc/secrets.yaml
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret notify_api_key retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret notify_user_key retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret homegear_host retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret pushover_api_key retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 DEBUG (SyncWorker_0) [homeassistant.util.yaml.loader] Secret pushover_user_key retrieved from secrets.yaml in folder /config
2022-07-10 22:04:34 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration browser_mod which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-10 22:04:34 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-10 22:04:34 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration mass which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-10 22:04:39 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Shield' for nfandroidtv integration not ready yet: Failed to connect to host: 192.168.1.31; Retrying in background
2022-07-10 22:04:46 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Shelly_Main_Bedroom' for shelly integration not ready yet: Timeout during device setup; Retrying in background
2022-07-10 22:05:12 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading configuration flow for integration mass: No module named 'music_assistant'
2022-07-10 22:07:39 ERROR (MainThread) [homeassistant.components.octoprint] Error requesting octoprint-cb7f9771aed8337642fa239a85ea92f5 data: Server disconnected

**Comment by NAVRockClimber** [comment]
I cranked up the logging level for HACS and re-installed MA (2022.6.10) I don't see any issues. Just a zip file that is downloaded that does not contain the library mentioned by you.

logger configuration:
```
logger:
  default: warn
  logs:
    custom_components.hacs: debug
    aiogithubapi: debug
```

logs:
```
2022-07-11 17:56:19 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Getting repository information
2022-07-11 17:56:20 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Running checks against 2022.6.10
2022-07-11 17:56:21 DEBUG (MainThread) [custom_components.hacs] <HacsData async_write> Saving data
2022-07-11 17:56:21 DEBUG (MainThread) [custom_components.hacs] <HACSStore async_save_to_store> Did not store data for 'hacs.hacs'. Content did not change
2022-07-11 17:56:22 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Getting repository information
2022-07-11 17:56:23 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Running checks against 2022.6.10
2022-07-11 17:56:24 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Running pre installation steps
2022-07-11 17:56:24 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Pre installation steps completed
2022-07-11 17:56:24 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Running installation steps
2022-07-11 17:56:24 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Getting repository information
2022-07-11 17:56:24 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Running checks against 2022.6.10
2022-07-11 17:56:25 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Local path is set to /config/custom_components/mass
2022-07-11 17:56:25 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Remote path is set to custom_components/mass
2022-07-11 17:56:25 DEBUG (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> ref: 2022.6.10 --- tag: 2022.6.10
2022-07-11 17:56:25 DEBUG (MainThread) [custom_components.hacs] <QueueManager> Checking out tasks to execute
2022-07-11 17:56:25 DEBUG (MainThread) [custom_components.hacs] <QueueManager> Starting queue execution for 1 tasks
2022-07-11 17:56:25 DEBUG (MainThread) [custom_components.hacs] Downloading https://github.com/music-assistant/hass-music-assistant/releases/download/2022.6.10/mass.zip
2022-07-11 17:56:26 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Download of mass.zip completed
2022-07-11 17:56:26 DEBUG (SyncWorker_17) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Cleaning up /tmp/tmpw5d14p29
2022-07-11 17:56:26 DEBUG (MainThread) [custom_components.hacs] <QueueManager> Queue execution finished for 1 tasks finished in 0.77 seconds
2022-07-11 17:56:26 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Installation steps completed
2022-07-11 17:56:26 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Running post installation steps
2022-07-11 17:56:26 INFO (MainThread) [custom_components.hacs] Reloading custom_component cache
2022-07-11 17:56:26 WARNING (SyncWorker_32) [homeassistant.loader] We found a custom integration browser_mod which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-11 17:56:26 WARNING (SyncWorker_21) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-11 17:56:26 WARNING (SyncWorker_24) [homeassistant.loader] We found a custom integration mass which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-07-11 17:56:26 INFO (MainThread) [custom_components.hacs] Custom_component cache reloaded
2022-07-11 17:56:26 INFO (MainThread) [custom_components.hacs] <Integration music-assistant/hass-music-assistant> Post installation steps completed
2022-07-11 17:56:26 DEBUG (MainThread) [custom_components.hacs] <HacsData async_write> Saving data
2022-07-11 17:56:26 DEBUG (MainThread) [custom_components.hacs] <HACSStore async_save_to_store> Did not store data for 'hacs.hacs'. Content did not change	
```

Is there a way to manually install the library as workaorund?

**Comment by NAVRockClimber** [comment]
Ok, after a little research I found a logger `homeassistant.util.package` a set the level for that one to debug as well.

Now I see the following lines in the log:

```
2022-07-11 18:47:41 INFO (SyncWorker_17) [homeassistant.util.package] Attempting install of music-assistant==1.4.11
2022-07-11 18:47:41 DEBUG (SyncWorker_17) [homeassistant.util.package] Running pip command: args=['/usr/local/bin/python3', '-m', 'pip', 'install', '--quiet', 'music-assistant==1.4.11', '--timeout', '60', '--upgrade', '--constraint', '/usr/src/homeassistant/homeassistant/package_constraints.txt', '--find-links', 'https://wheels.home-assistant.io/musllinux/', '--prefer-binary', '--user', '--prefix=']
2022-07-11 18:47:49 ERROR (MainThread) [homeassistant.config_entries] Error occurred loading configuration flow for integration mass: No module named 'music_assistant'
```

**Comment by erkr** [comment]
@NAVRockClimber thx!

**Comment by NAVRockClimber** [comment]
Yes, it worked but I would not recommend this to unexperienced users.

**Comment by erkr** [comment]
Hi Marcel. Although 'fixed' via manual library install. What about the root cause?
Keep open as a bug?!

**Comment by marcelveldt** [comment]
> Hi @marcelveldt. Although 'fixed' via manual library install. What about the root cause? Keep open as a bug?!

No this is not a MA bug but a conflict in dependencies which is a common HA issue, especially when custom components get involved. The only thing I can think off is a warning in our docs that other Youtube integrations can not be used side-by-side with Music Assistant.

**Comment by OzGav** [comment]
> The only thing I can think off is a warning in our docs that other Youtube integrations can not be used side-by-side with Music Assistant.
I have added a NOTE at the top of the YouTube How Do I?

**Comment by erkr** [comment]
Time to close the issue then, @OzGav thanks

**Comment by NAVRockClimber** [comment]
> > Hi @marcelveldt. Although 'fixed' via manual library install. What about the root cause? Keep open as a bug?!
> 
> No this is not a MA bug but a conflict in dependencies which is a common HA issue, especially when custom components get involved. The only thing I can think off is a warning in our docs that other Youtube integrations can not be used side-by-side with Music Assistant.

@marcelveldt and @OzGav why do you think I got another youtube integration installed? I think you confusing this issue with another. I got no youtube extension installed. In fact this was the reason to install MA at the first place.

/reopen

**Comment by erkr** [comment]
Thanks for this update

**Comment by NAVRockClimber** [comment]
@erkr thanks for reopening this issue. 

What I can do to verify your suspicion is to setup a clean container and check if one of my integrations interferes with MA. But this is probably quite time consuming and unfortunately I am this week on a tight schedule. It might be that it will get next week before I get to testing this.

**Comment by marcelveldt** [comment]
@NAVRockClimber I assumed Youtube because so far that has been the only conflict we've seen but there may be others.
Installing Music Assistant on a "virgin" HA install with only core integrations should always work 100% ad that is extensively tested. If other custom integrations are added it will become more hit and miss because those custom integrations are never tested against eachother. 

I think this will work in 95% of the cases btw and only sometimes can cause a collision

**Comment by marcelveldt** [comment]
It would still be nice though to find out what other integration caused the collision.

**Comment by NAVRockClimber** [comment]
Ok, I just sat down and gave it try. It happens on a clean kubernetes container (new empty volume) as well. 

- A pure basic installation as if you would start from the scratch with HA
- installed HACS via shell, restartet
- configured HACS, restarted
- installed MA via HACS, restarted, 
- configured integration and got the same error message.

But the library appears. It seems to be something after installing the library:

![image](https://user-images.githubusercontent.com/37535503/180860669-49d03dd8-a076-49e5-be41-99b81221bf53.png)

My first idea was that the problem is caused by some differences in the file permission handling in kubernetes but due to the library seems to be present I am a little bit out of ideas.

**Comment by erkr** [comment]
Thanks for the efforts made to provide this additional information!

**Comment by marcelveldt** [comment]
strange question perhaps but why is all logging duplicated ?

what hardware are you running ? Just x86-64 based ?

It is really strange because multiple users are running the Home Assistant Container theirselves and have no issues with MA at all. You said you're running the official HA image, only difference is that it is a Kubernetes cluster and not plain/standard docker. That's as far as my knowledge of docker/kubernetes goes so hopefully someone is able to understand what is going here because I do not.

Music Assistant is tested and confirmed to be working on:
Home Assistant OS
Home Assistant Container / Supervised

When running HA Core yourself you are kind of on your own because I have no clue what OS level packages are then missing etc. Hopefully someone in the community can jump in at this point

**Comment by NAVRockClimber** [comment]
> strange question perhaps but why is all logging duplicated ?

Not sure anymore what I did there. I was already exhausted from the day. I just saw after installing a extension in hacs it seems to re-init some things and adds a few lines

> what hardware are you running ? Just x86-64 based ?

It is a mixed setup of spare components I threw togehter about 1.5 years ago. 6 RPis 4 and 2 x86-64. 
In my test yesterday the container was scheduled on a x86 node.

I just tested it on another Pi I had laying around and spun up clean container in docker (just a volume and publishing port 8123) and everthing works there. I gues it is somehow K8S related. 

Not really sure what the difference is but I will try the next days to force the scheduling on a pi. Another thing I could try is replacing the NFS storage by something else. 

But I think this is quite a corner case.
